src/routes/safe/components/Apps/components/SafeAppCard/SafeAppCard.tsx

```diff
@@ -0,0 +1,238 @@
+import enqueueSnackbar from 'src/logic/notifications/store/actions/enqueueSnackbar'
+import { enhanceSnackbarForAction, NOTIFICATIONS } from 'src/logic/notifications'
+import { FETCH_STATUS } from 'src/utils/requests'
+import { copyToClipboard } from 'src/utils/clipboard'
+import { getShortName } from 'src/config'
+import { SafeAppDescriptionSK, SafeAppLogoSK, SafeAppTitleSK } from './SafeAppSkeleton'
+
+type SafeAppCardSize = 'md' | 'lg'
+
+type SafeAppCardProps = {
+  safeApp: SafeApp
+  size: SafeAppCardSize
+  togglePin: (app: SafeApp) => void
+  isPinned?: boolean
+  isCustomSafeApp?: boolean
+  onRemove?: (app: SafeApp) => void
+}
+
+const SafeAppCard = ({
+  safeApp,
+  size,
+  togglePin,
+  isPinned,
+  isCustomSafeApp,
+  onRemove,
+}: SafeAppCardProps): React.ReactElement => {
+  const chainId = useSelector(currentChainId)
+  const dispatch = useDispatch()
+
+  const safeAddress = extractSafeAddress()
+  const appsPath = generateSafeRoute(SAFE_ROUTES.APPS, {
+    shortName: getShortName(),
+    safeAddress,
+  })
+  const openSafeAppLink = `${appsPath}?appUrl=${encodeURI(safeApp.url)}`
+
+  const shareSafeApp = () => {
+    const shareSafeAppUrl = getShareSafeAppUrl(safeApp.url, chainId)
+    copyToClipboard(shareSafeAppUrl)
+    dispatch(enqueueSnackbar(enhanceSnackbarForAction(NOTIFICATIONS.SHARE_SAFE_APP_URL_COPIED)))
+  }
+
+  const isSafeAppLoading = safeApp.fetchStatus === FETCH_STATUS.LOADING
+
+  if (isSafeAppLoading) {
+    return (
+      <SafeAppContainer size={size} layout initial={{ opacity: 0 }} animate={{ opacity: 1 }} exit={{ opacity: 0 }}>
+        <StyledAppCard size={size}>
+          <LogoContainer size={size}>
+            <SafeAppLogoSK size={size} />
+          </LogoContainer>
+          <DescriptionContainer size={size}>
+            <SafeAppTitleSK />
+            <SafeAppDescriptionSK />
+            <SafeAppDescriptionSK />
+          </DescriptionContainer>
+        </StyledAppCard>
+      </SafeAppContainer>
+    )
+  }
+
+  return (
+    <SafeAppContainer size={size} layout initial={{ opacity: 0 }} animate={{ opacity: 1 }} exit={{ opacity: 0 }}>
+      <StyledLink to={openSafeAppLink} aria-label={`open ${safeApp.name} Safe App`}>
+        <StyledAppCard size={size}>
+          {/* Safe App Logo */}
+          <LogoContainer size={size}>
+            <SafeAppLogo
+              size={size}
+              src={safeApp.iconUrl}
+              alt={`${safeApp.name || 'Safe App'} Logo`}
+              onError={setSafeAppLogoFallback}
+            />
+          </LogoContainer>
+
+          {/* Safe App Description */}
+          <DescriptionContainer size={size}>
+            <SafeAppTitle size="xs">{safeApp.name}</SafeAppTitle>
+            <SafeAppDescription size="lg">{safeApp.description}</SafeAppDescription>
+          </DescriptionContainer>
+
+          {/* Safe App Actions */}
+          <ActionsContainer onClick={(e) => e.preventDefault()}>
```

Comment:
Can you not `stopPropagation` in the handler that causes the need to prevent this?

Reply:
I think that the best option here is to keep as it is. Otherwise we will need to inlcude the `stopPropagation()` in all the `shareSafeApp`, `togglePin`,  `onRemove` callbacks. (and in the future, if we add a new action, we might accidentally forget to add the `stopPropagation()` on it)

Reply:
Alright, I didn't realise that it goverened a few functions. It would be good to add a comment explaining this.

Reply:
I think it is not necessary, it is clear enough.

Reply:
On face value, it's almost an empty function but if you don't think it's necessary, I'll resolve this.